File: src/lib/types/day-attempt.ts
export interface DayAttempt {
	id: string;
	day_plan_id: string;
	day_plan_version: number;
	attempt_number: number;
	status: 'in_progress' | 'submitted' | 'blocked' | 'passed' | 'mastery';
	score_implementation: number;
	score_code_quality: number;
	score_accessibility: number;
	score_performance: number;
	score_quiz: number;
	total_score: number;
	memory_rebuild_completed: number;
	memory_rebuild_passed: number;
	memory_rebuild_notes: string;
	what_broke: string;
	why_broke: string;
	how_fixed: string;
	refactor_tomorrow: string;
	daily_summary: string;
	exercise_notes: string;
	code_snapshot: string;
	started_at: string;
	submitted_at: string | null;
	actual_minutes: number;
	is_draft: number;
	last_autosave: string;
	created_at: string;
	updated_at: string;
}

export interface UpdateAttemptInput {
	status?: string;
	score_implementation?: number;
	score_code_quality?: number;
	score_accessibility?: number;
	score_performance?: number;
	score_quiz?: number;
	memory_rebuild_completed?: number;
	memory_rebuild_passed?: number;
	memory_rebuild_notes?: string;
	what_broke?: string;
	why_broke?: string;
	how_fixed?: string;
	refactor_tomorrow?: string;
	daily_summary?: string;
	exercise_notes?: string;
	code_snapshot?: string;
	actual_minutes?: number;
}

File: src-tauri/src/db/models/day_attempt.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct DayAttempt {
    pub id: String,
    pub day_plan_id: String,
    pub attempt_number: i32,
    pub status: String,
    pub is_draft: bool,
    pub score_implementation: i32,
    pub score_code_quality: i32,
    pub score_accessibility: i32,
    pub score_performance: i32,
    pub score_quiz: i32,
    pub total_score: i32,
    pub daily_summary: String,
    pub what_went_well: String,
    pub what_to_improve: String,
    pub key_learnings: String,
    pub memory_rebuild_passed: bool,
    pub memory_rebuild_notes: String,
    pub actual_minutes: i32,
    pub started_at: String,
    pub submitted_at: Option<String>,
    pub last_autosave: String,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateDayAttemptInput {
    pub day_plan_id: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct UpdateDayAttemptInput {
    pub daily_summary: Option<String>,
    pub what_went_well: Option<String>,
    pub what_to_improve: Option<String>,
    pub key_learnings: Option<String>,
    pub memory_rebuild_passed: Option<bool>,
    pub memory_rebuild_notes: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SubmitScoresInput {
    pub score_implementation: i32,
    pub score_code_quality: i32,
    pub score_accessibility: i32,
    pub score_performance: i32,
    pub score_quiz: i32,
    pub daily_summary: String,
    pub what_went_well: String,
    pub what_to_improve: String,
    pub key_learnings: String,
    pub memory_rebuild_passed: bool,
    pub memory_rebuild_notes: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct DayAttemptSummary {
    pub id: String,
    pub day_plan_id: String,
    pub day_title: String,
    pub day_number: i32,
    pub attempt_number: i32,
    pub status: String,
    pub total_score: i32,
    pub memory_rebuild_passed: bool,
    pub actual_minutes: i32,
    pub submitted_at: Option<String>,
    pub created_at: String,
}
